### DISCUSSION TITLE
GitHub Actions usage

### DISCUSSION THREAD

**Discussion by foarsitter** [cookiejar/cookietemple#726] [General]
First of all, thanks for the effort you you all have put in to this! 

Within minutes I was up and running with all the bells and whistles a developer can wish but within minutes I was out of those minutes GitHub gave me for free. 

I made the mistake to create a private repository with the template. Switching to a public repo was fine for this project but a private project will soon get an expensive one I guess. 

What are your thoughts about this? 

Of course this are the costs of modern developments but I wasn't expect to burn my 2000 minutes in a little more of 2 hours.

**Comment by Zethson** [comment]
Hi @foarsitter,

great to see that it worked for you. I perfectly agree that this is an issue.

I think that we have 2 primary sources of this issue that we could maybe try to do something about:

1. Dependabot. If you have dependabot enabled you will immediately get spammed by many PRs which all trigger GA and your minutes are gone. I honestly don't know how we can solve the dependabot issue besides limiting the dependabot PRs per week to even less than 3.
2. Testing for multiple Python versions. Since we are using matrices for the tests if you are testing multiple Python versions the minutes are also gone very quickly. I think that we can fix this to maybe 3.9 or something and just not test the other versions from the get to. If users want that they have to adapt it themselves.

**Reply by foarsitter** [reply]
Maybe a warning for private projects could be enough? Having all those extra's is a plus of this template and people new to this template are only subject to this issue. The second time I will start a new project I will be aware of whats happening when the actions are triggered.

**Reply by Zethson** [reply]
Yeah. We can add a warning. 
Are you interested in filing up a PR? Just a print (bold & yellow) with the text above https://github.com/cookiejar/cookietemple/blob/master/cookietemple/create/github_support.py#L267

**Reply by foarsitter** [reply]
#729 is there to address this issue
